pyprland/plugins/scratchpads.py

```diff
@@ -394,22 +391,46 @@ async def run_show(self, uid, force=False) -> None:
 
         await hyprctl(f"focuswindow {addr}")
 
+        size = self._convert_coords(item.conf.get("size"), monitor)
         if size:
-            # NOTE: Format for size is "X_SIZE Y_SIZE"
-            # X_SIZE, Y_SIZE is percentage of monitor size
-            x_size_p, y_size_p = map(int, size.split())
-            x_size, y_size = int(monitor["width"] * x_size_p / 100), int(monitor["height"] * y_size_p / 100)
-
+            x_size, y_size = size
             await hyprctl(f"resizewindowpixel exact {x_size} {y_size},{addr}")
 
+        position = self._convert_coords(item.conf.get("position"), monitor)
         if position:
-            # NOTE: Format for position is "X_POS Y_POS"
-            # X_POS, Y_POS is percentage of monitor size from top left corner
-            x_pos_p, y_pos_p = map(int, position.split())
-            x_pos, y_pos = int(monitor["width"] * x_pos_p / 100), int(monitor["height"] * y_pos_p / 100)
+            x_pos, y_pos = position
             x_pos_abs, y_pos_abs = x_pos + monitor["x"], y_pos + monitor["y"]
-
             await hyprctl(f"movewindowpixel exact {x_pos_abs} {y_pos_abs},{addr}")
 
         await asyncio.sleep(0.2)  # ensure some time for events to propagate
         self.transitioning_scratches.discard(uid)
+
+    def _convert_coords(self, coords, monitor):
+        """
+        Converts a string like "X Y" to coordinates relative to monitor
+        Supported formats for X, Y:
+        - Percentage: "V%". V in [0; 100]
+
+        Example:
+        "10% 20%", monitor 800x600 => 80, 120
+        """
+
+        if not coords:
```

Comment:
Can we show an error log here ?

Reply:
Or just an assert instead of those two lines...

Reply:
I've reworked `_convert_coords` a bit. Now we check presence of `scale` and `position` in config before calling it.